### PR TITLE
EDMFX bomex box test in the CI (replacing sphere test)

### PR DESCRIPTION
**Pull request by trontrytel**
Replacing EDMFX Bomex test on a sphere with EDMFX Bomex test in a box

### CODE REVIEW

**Comment by trontrytel** [comment]
@szy21 - could you check if those flags and test cases make sense for you in the CI?

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -461,60 +461,87 @@ steps:
...
 
+      - label: ":genie: EDMFX Bomex in a box"
+        command: >
+          julia --color=yes --project=examples examples/hybrid/driver.jl --job_id edmfx_bomex_box
+          --initial_condition Bomex --subsidence Bomex --edmf_coriolis Bomex --ls_adv Bomex --surface_setup Bomex
+          --turbconv edmfx --edmfx_entr_detr true --edmfx_sgs_flux true
+          --moist equil
+          --config box --hyperdiff true --kappa_4 1e8
+          --x_max 1e5 --y_max 1e5 --x_elem 2 --y_elem 2 --z_elem 60 --z_max 3e3 --z_stretch false
+          --dt 10secs --t_end 1hours --dt_save_to_disk 5mins
```
Comment (szy21) [inline .buildkite/pipeline.yml:513]: dt = 10secs may be too large if you have, say, updraft velocity > 5m/s and dz = 50m here.

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -461,60 +461,87 @@ steps:
...
 
-      - label: ":genie: EDMFX Bomex on a sphere"
-        command:
-          - julia --color=yes --project=examples examples/hybrid/driver.jl --job_id edmfx_bomex_sphere --ode_algo SSP33ShuOsher --apply_limiter false --initial_condition Bomex --moist equil --turbconv edmfx --subsidence Bomex --edmf_coriolis Bomex --ls_adv Bomex --surface_setup Bomex --z_elem 60 --z_stretch false --z_max 3e3 --dt 0.001secs --t_end 0.1secs --dt_save_to_disk 0.01secs --reference_job_id edmf_bomex
-        artifact_paths: "edmfx_bomex_sphere/*"
+      - label: ":genie: Diagnostic EDMFX Bomex in a box"
+        command: >
+          julia --color=yes --project=examples examples/hybrid/driver.jl --job_id diagnostic_edmfx_bomex_box
+          --initial_condition Bomex --subsidence Bomex --edmf_coriolis Bomex --ls_adv Bomex --surface_setup Bomex
+          --turbconv diagnostic_edmfx --edmfx_entr_detr true
```
Comment (szy21) [inline .buildkite/pipeline.yml:481]:
```suggestion
          --turbconv diagnostic_edmfx
```
(We don't have entrainment for diagnostic edmfx in the main yet, so this argument won't do anything. Maybe delete it to be clear?)

**Review by szy21 — APPROVED** [review]
Looks good, thanks!

**Pull request merged**